src/librbd/AioObjectRequest.h

```diff
@@ -348,10 +348,6 @@ class AioObjectTrim : public AbstractAioObjectWrite {
   virtual void pre_object_map_update(uint8_t *new_state) {
     *new_state = OBJECT_PENDING;
   }
-
-  virtual bool post_object_map_update() {
```

Comment:
I didn't realize that the trim state machine already performed a batch update -- I think this removal is correct and is what was (incorrectly) causing per-object object map updates.